infrastructure/src/main/java/org/corfudb/infrastructure/datastore/DataStore.java

```diff
@@ -94,11 +94,14 @@ public DataStore(@Nonnull Map<String, Object> opts,
     }
 
 
-    public static int getChecksum(byte[] bytes) {
+    private static int getChecksum(ByteBuffer buf) {
         Hasher hasher = Hashing.crc32c().newHasher();
-        for (byte a : bytes) {
-            hasher.putByte(a);
+
+        buf.mark();
```

Comment:
Can you please add a test for this -- ensure that the buffer pointer remains the same. Also I do not see any tests that exercise this functionality.